### PR TITLE
Fixing the accuracy of values

### PR DESCRIPTION
**Pull request by hupe13**
Hi Raruto,
I'm made a pull request. I would like to show you the differences here, so you can see what effects it has. 
The tracks are a hike 18 km, 
grafik
a flight,
grafik
a hike with a very difficult climb and many measurement errors, 
grafik
and a run..
grafik
I'm unable to program the unit of pace (min/km)  correctly, then the information on the y axis is not correct.
Please have a look at it, change the code if necessary.

I documented the changes [here](https://github.com/hupe13/extensions-leaflet-map-github/blob/main/leaflet-plugins/hupe13_changes-elevation.txt).

### CODE REVIEW

File: libs/leaflet-gpxgroup.js [hunk 1]
```diff
@@ -164,7 +164,7 @@ L.GpxGroup = L.Class.extend({
       .then(text => this._elevation._parseFromString(text))
       .then(geojson => {
         if(geojson) {
-          geojson.name = geojson.name || (geojson[0] && geojson[0].properties.name) || track.split('/').pop().split('#')[0].split('?')[0];
+          geojson.name = this.options.filename ? track.split('/').pop().split('.').slice(0, -1).join('.') : (geojson.name || (geojson[0] && geojson[0].properties.name) || track.split('/').pop().split('.').slice(0, -1).join('.'));
```
Comment (Raruto) [inline libs/leaflet-gpxgroup.js:167]: I think this has already been discussed here https://github.com/Raruto/leaflet-elevation/pull/160, anyway, it is beyond the scope of the current pull

File: package-lock.json [hunk 2]
```diff
@@ -55,6 +55,64 @@
 			"integrity": "sha512-3U9y+43hz7ZM+rzG24Qe2mufW5KhvFg/NhnNph+i9mgCtdTCtMJuI1TMkrIUiK7Ix4PYlRF9I5dhqaLYA/ADXw==",
```
Comment (Raruto) [inline package-lock.json:55]: This sounds strange, why are you submitting an edited `package-lock.json` file?

File: src/utils.js [hunk 3]
```diff
@@ -131,9 +133,10 @@ export const randomId      = ()                   => Math.random().toString(36).
 /**
  * TODO: use generators instead? (ie. "yield")
  */
-export const iMax = (iVal, max = -Infinity) => (iVal > max ? iVal : max); 
-export const iMin = (iVal, min = +Infinity) => (iVal < min ? iVal : min);
-export const iAvg = (iVal, avg = 0, idx = 1) => (iVal + avg * (idx - 1)) / idx;
+export const iMax = (iVal, max = -Infinity) => (iVal > max ? iVal : max);
+export const iMin = (iVal, min = +Infinity) => ((iVal && iVal < min) ? iVal : min );
+export const iMinP = (iVal, min = +Infinity) => ((iVal && iVal < min && iVal > 0) ? iVal : min );
```
Comment (Raruto) [inline src/utils.js:138]: a shiny new ambiguous function? (why do you need a new one?)

File: src/utils.js [hunk 4]
```diff
@@ -156,4 +159,13 @@ export const clamp     = (val, range)           => range ? (val < range[0] ? ran
...
+export const wrapDelta = (curr, prev, deltaMax) => Math.abs(curr - prev) > deltaMax ? prev + deltaMax * Math.sign(curr - prev) : curr;
+
+/**
+ * Round numbers according to size
+ */
+export function tooltipvalue(value,count){
+	let comma = count - 1 - Math.floor(Math.log10(Math.abs(value)));
+	let factor = Math.pow(10,comma);
+	return Math.round(value * factor) / factor;
+}
```
Comment (Raruto) [inline src/utils.js:171]: A new function that is hard to debug, you can include some [tests](https://github.com/Raruto/leaflet-elevation/blob/3ee745d71ed172979b7bfe0ff52f7c63214d3c04/spec/utils.test.js) to explain better your intents..

File: src/utils.js [hunk 5]
```diff
@@ -27,6 +27,8 @@ export function formatTime(t) {
 	if ( s === 60 ) { m++; s = 0; }
 	if ( m === 60 ) { h++; m = 0; }
 	if ( h === 24 ) { d++; h = 0; }
+	if ( !d && !h && !m ) return s.toString().padStart(2, 0) + '"';
+	if ( !d && !h ) return m.toString().padStart(2, 0) + "'" + s.toString().padStart(2, 0) + '"';
 	return (d ? d + "d " : '') + h.toString().padStart(2, 0) + ':' + m.toString().padStart(2, 0) + "'" + s.toString().padStart(2, 0) + '"';
```
Comment (Raruto) [inline src/utils.js:32]: what we are trying to solve here?

**Review by Raruto — COMMENTED** [review]
Hi hupe, it seems you want to solve too many things in one single pull, for ease of all it is if you separate your changes into several smaller pull requests.

It's never too late to add some shiny new bugs 😄

File: package-lock.json (hunk 2 above)
Comment (hupe13) [reply package-lock.json:55]: Ignore this.

File: src/utils.js (hunk 3 above)
Comment (hupe13) [reply src/utils.js:138]: There is a minimum < 0 (like for slope e.g.) and a minimum > 0 (like for speed and pace).

File: src/utils.js (hunk 4 above)
Comment (hupe13) [reply src/utils.js:171]: https://leafext.info/javatest.php

Comment (Raruto) [reply src/utils.js:171]: @hupe13 are you looking for something like that?

```js
let round = (x, n) => +x.toPrecision(n)
```

**Ref:** [How can I round to an arbitrary number of significant digits with JavaScript?](https://stackoverflow.com/questions/36369239/how-can-i-round-to-an-arbitrary-number-of-significant-digits-with-javascript)

File: package-lock.json (hunk 2 above)
Comment (Raruto) [reply package-lock.json:55]: ok, but please update the pull code (I won't do it for you ...)

**Comment by hupe13** [comment]
I will close the pull request and open a separate one for each fix. I found some things that I do differently now and you have published a new version.

File: src/utils.js (hunk 4 above)
Comment (hupe13) [reply src/utils.js:171]: Thank you very much! This works better than my code!
https://leafext.info/javatest.php

**Pull request closed without merging**